docs/install-and-setup.md

```diff
@@ -210,12 +210,25 @@ To set up command-line completion, source the output of
 `jj util completion bash/zsh/fish`. Exactly how to source it
 depends on your shell.
 
+Improved completions are currently in the works, these will complete things
+like bookmark names as well. You can activate them with the alternative "dynamic"
+instructions below. Please let us know if you encounter any issues, so we can
+ensure a smooth transition once we default to these new completions. If you
+have ideas about specific completions that could be added, please share them
+[here](https://github.com/martinvonz/jj/issues/4763).
+
 ### Bash
 
 ```shell
 source <(jj util completion bash)
 ```
 
+dynamic:
+
+```shell
+echo "source <(COMPLETE=bash jj)" >> ~/.bashrc
```

Comment:
Is there a reason the user would want non-dynamic completions only in the current shell but dynamic completions in future shells too by updating their shell config? If there wasn't a real reason for that, I think we should align the two. I don't feel strongly about which form we should choose. Updating the shell config is probably what most users want. Do we need to worry about users running this line multiple times and ending up with a cluttered config?

Reply:
Another thing that it'd be nice to make clearer is whether the user wants both (which would be my guess) or just the dynamic one.

Assuming it's both, I'd do something like:

> Add these lines to `~/.bashrc`:
>
> ```shell
> source <(jj util completion bash)
> source <(COMPLETE=bash jj)  # If you also want dynamic completions
> ```

I think adding the line about adding these to .bashrc is a good enough and less noisy substitute compared to including `echo ... > ~/.bashrc`, but there's also an argument of having a copy-pastable command; either is OK by me. It's more important to me that the two are aligned.

Reply:
> Another thing that it'd be nice to make clearer is whether the user wants both (which would be my guess) or just the dynamic one.

That's what I thought too, but the instructions above say this (emphasis mine):

> You can activate them with the *alternative* "dynamic" instructions below.

@senekor, is there any reason to use the static version? Is it just that we're not confident enough in the dynamic ones yet?

Reply:
I think they're just copied from https://docs.rs/clap_complete/latest/clap_complete/env/index.html

Dropping `echo ".." >> ..` makes sense.

Reply:
I opened a [PR](https://github.com/martinvonz/jj/pull/4847/files) to address this.

I did indeed copy these without thinking too much. People may have different preferences about organizing their shell configs. When I used to use zsh, I had the init script configured as `~/.config/zsh/.zshrc` or something, to declutter my home directory. For fish I like to split up my config in many small files. So I agree with removing that part.

> Another thing that it'd be nice to make clearer is whether the user wants both (which would be my guess) or just the dynamic one.

The dynamic completions should already do all the things the static ones did. So there's no need to have both. I added a sentence in the PR mentioning that.

> is there any reason to use the static version? Is it just that we're not confident enough in the dynamic ones yet?

Exactly. Because the dynamic completions are supposed to _replace_ the static ones, I'm not entirely sure if there are some regressions which don't apply to my workflow.

Reply:
> we're not confident enough in the dynamic ones yet?

I tried dynamic completion on zsh, and noticed that `jj <TAB>` only lists "log". Apparently, zsh completion script relies on `_CLAP_COMPLETE_INDEX=$(expr $CURRENT - 1)`, which would get out of sync after alias/default-command substitution?

Reply:
Thanks for checking. Sounds like we should not recommend only dynamic completions yet then.